src/Microsoft.Health.Fhir.Shared.Core/Features/Resources/Upsert/UpsertResourceHandler.cs

```diff
@@ -61,7 +61,7 @@ public async Task<UpsertResourceResponse> Handle(UpsertResourceRequest message,
             bool allowCreate = await ConformanceProvider.Value.CanUpdateCreate(resource.TypeName, cancellationToken);
             bool keepHistory = await ConformanceProvider.Value.CanKeepHistory(resource.TypeName, cancellationToken);
 
-            ResourceWrapper resourceWrapper = CreateResourceWrapper(resource, deleted: false);
+            ResourceWrapper resourceWrapper = CreateResourceWrapper(resource, deleted: false, keepMeta: false);
```

Comment:
In CosmosDb the new version is determined in the Stored Proc, there will likely need to be some updates here too: https://github.com/microsoft/fhir-server/blob/master/src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/StoredProcedures/Upsert/upsertWithHistory.js#L98

Reply:
I'm only setting the VersionId for creates, not upserts. It doesn't seem like there are any necessary changes in the create case.  For creates that come in on the upsert endpoint, we would still do the deserialize/serialize to update the versionId and lastUpdated properties.

Reply:
I've applied John's suggestion of assuming Version = 1 on upserts, and updated the stored proc to set the rawResourceMetaSet flag to false in the stored proc when the version isn't 1.